### ISSUE TITLE
Feature request: either obey XDG fs paths, or let us override MACKUP_CONFIG_FILE, CUSTOM_APPS_DIR

### ISSUE THREAD

**Issue by kergoth** [lra/mackup#473]
See http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html. ~/.config/mackup/* would be ideal, rather than cluttering up the homedir with yet another dotfile.

**Comment by mkwmms** [comment]
huge +1

**Comment by lra** [comment]
So mackup should look at the `$XDG_CONFIG_HOME` env var and try to look for the `.mackup.cfg` and all the custom application files? All in addition to the usual `$HOME` place?

Makes sense?

**Comment by mkwmms** [comment]
It would be either/or:
So if `$XDG_CONFIG_HOME` is set it would look at: `$XDG_CONFIG_HOME/mackup/mackup.cfg` (notice these files/dirs are not invisible), if it is not set, then it will look at: `$HOME/.mackup.cfg`. I'm wondering how the `~/.mackup` directory (for custom app.cfg files or to override default app.cfg files) would be represented as well... Maybe they would just be in `$XDG_CONFIG_HOME/mackup` alongside `mackup.cfg`?

See the `--global` option under the [git docs](http://git-scm.com/docs/git-config) for a good example of how they obey the XDG spec.

**Comment by lra** [comment]
Ok then:

1. try `$XDG_CONFIG_HOME/mackup/` and `$XDG_CONFIG_HOME/mackup.cfg` if they exist.
1. if not, try `$HOME/.mackup/` and `$HOME/.mackup.cfg`
1. if not, no config.

Make sense?

**Comment by kergoth** [comment]
Sounds reasonable to me.

**Comment by mkwmms** [comment]
It is customary for every app/package to have their own directory in $XDG_CONFIG_HOME (or any other XDG directory), so it would be:
1. try $XDG_CONFIG_HOME/mackup/ and $XDG_CONFIG_HOME/mackup/mackup.cfg if they exist.
2. if not, try $HOME/.mackup/ and $HOME/.mackup.cfg
3. if not, no config.

I figure might as well be picky now before we add the feature rather than later...

I love this package by the way. Nice work!

**Comment by lra** [comment]
Can't we have 2 things: the mackup config `.mackup.cfg` and the custom application database, `mackup/*`. But in the custom application database, there can be one for mackup, so in your suggestion, `mackup/mackup.cfg` would lead to bothe the mackup config, and the application database entry for mackup.

**Comment by mkwmms** [comment]
Am I understanding you correctly that, as it stands right now, you can have `$HOME/.mackup.cfg` and `$HOME/mackup/.mackup.cfg` aka a mackup entry in the custom apps database? 
Right now does it:
1. try: `$HOME/.mackup/mackup.cfg`
2. if not, try:`$HOME/.mackup.cfg`
3. if not, no config.

If you want to keep it consistent between XDG and non XDG, it could behave like your original suggestion:
1. try `$XDG_CONFIG_HOME/mackup/` and `$XDG_CONFIG_HOME/mackup.cfg` if they exist.
2. if not, try `$HOME/.mackup/` and `$HOME/.mackup.cfg`
3. if not, no config.

**Comment by lra** [comment]
No, `$HOME/.mackup/mackup.cfg` and `$HOME/.mackup.cfg` are 2 different thins.
The former is a custom app database entry for mackup, the later is the mackup config.

**Comment by mkwmms** [comment]
Ok that makes a lot more sense.
Then you original suggestion: 

1. try `$XDG_CONFIG_HOME/mackup/` and `$XDG_CONFIG_HOME/mackup.cfg` if they exist.
2. if not, try `$HOME/.mackup/` and `$HOME/.mackup.cfg`
3. if not, no config.

is definitely the way to go.

**Comment by lra** [comment]
ok let's do it then, PR welcome ;)

**Comment by mkwmms** [comment]
Awesome, Ill see if I can do a PR soon

**Comment by NorthIsUp** [comment]
added this #1987
